### ISSUE TITLE
Discus

### ISSUE THREAD

**Issue by c835722** [ppoffice/hexo-theme-icarus#330]
Please make sure these boxes are checked before submitting your issue. Thank you!

- [X] I have setuped and configurated the blog according to [Hexo official documentation](https://hexo.io/);
- [X] I have read the [Theme Wiki](https://github.com/ppoffice/hexo-theme-icarus/wiki) carefully and created my own configuration file(_config.yml);
- [X] I have looked up the [Issues](https://github.com/ppoffice/hexo-theme-icarus/issues) and found no duplicate issues.

请在发布新Issue之前确保你已经进行了如下工作，谢谢！

- [ ] 我已经按照[Hexo官方文档](https://hexo.io/)中的步骤安装与配置了Hexo；
- [ ] 我已经仔细地阅读了[主题Wiki](https://github.com/ppoffice/hexo-theme-icarus/wiki)并创建了自己的配置文件(_config.yml)；
- [ ] 我已经搜索了[Issues](https://github.com/ppoffice/hexo-theme-icarus/issues)，并且没有找到类似的问题。

Hi,
Thank you for this project which I have used successfully.
I am trying to configure comments via Disqus for which I have an account and shortname configured.
My Icarus configuration is as follows in _config.yml
```
comment:
    disqus: in_training
```
I then hexo clean; hexo generate; hexo server.

When the site is generated and I click on the comments button on a post I get a reference generated as follows:

```
http://localhost:4000/2016/01/01/kickoff/#comments 
```
which then does not resolve. HTTP 404.

My configuration is as follows:
```
hexo version
hexo: 3.7.1
hexo-cli: 1.1.0
os: Darwin 17.5.0 darwin x64
http_parser: 2.8.0
node: 10.4.0
v8: 6.7.288.43-node.7
uv: 1.20.3
zlib: 1.2.11
ares: 1.14.0
modules: 64
nghttp2: 1.29.0
napi: 3
openssl: 1.1.0h
icu: 61.1
unicode: 10.0
cldr: 33.0
tz: 2018c
```

My Hexo configuration (key component only)
```
# Directory
source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:

# Writing
new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link: true # Open external links in new tab
filename_case: 0
render_drafts: false
post_asset_folder: true
relative_link: false
future: true
highlight:
  enable: true
  line_number: true
  auto_detect: false
  tab_replace:

# Category & Tag
default_category: uncategorized
category_map:
tag_map:

# Date / Time format
## Hexo uses Moment.js to parse and display date
## You can customize the date format as defined in
## http://momentjs.com/docs/#/displaying/format/
date_format: YYYY-MM-DD
time_format: HH:mm:ss

# Pagination
## Set per_page to 0 to disable pagination
per_page: 10
pagination_dir: page

# Extensions
## Plugins: https://hexo.io/plugins/
## Themes: https://hexo.io/themes/
theme: icarus

# Deployment
## Docs: https://hexo.io/docs/deployment.html
deploy:
  type:
```

My objective is to have Disqus comments working per post.
Is the possible currently?

**Comment by ppoffice** [comment]
@c835722 Did you set these two?

https://github.com/ppoffice/hexo-theme-icarus/blob/site/_config.yml#L15-L16

**Comment by c835722** [comment]
I did not have root set which I have adjusted.
Whilst this allows the individual page to resolve it doesn't change the original issue in my opinion.
What I am trying to achieve is getting the Disqus feature of commenting working with the plugin.
Could you perhaps point me at a working example that uses Disqus so that I can try to copy the configuration?
Thank you for the assistance.

**Comment by ppoffice** [comment]
@c835722 Disqus is working per post, as you can see here: https://github.com/ppoffice/hexo-theme-icarus/blob/master/layout/comment/disqus.ejs#L7

I still don't get your problem. Is Disqus not working at all? Or is the comment link failed to resolve? Or  are you getting same comments for different posts?

**Comment by c835722** [comment]
Thank you for your assistance @ppoffice .

I had configured the hexo/icarus site _config.yml to use a namespace.gitlab.io/blog page.
This was the configuration that had created the issue for me where Disqus was not working.

FIX:
When I CNAMEd a new personal hostname.domain.name to the namespace.gitlab.io site the Disqus extension worked as specified when entering hostname.domain.name into a Browser.
For this configuration I set the url to hostname.domain.name and the root to / in hexo _config.yml
